sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/application-changes-report-integration/_tests_/application-changes-report-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,383 @@
+import { DeepMocked } from "@golevelup/ts-jest";
+import { INestApplication } from "@nestjs/common";
+import {
+  QueueNames,
+  addDays,
+  getISODateOnlyString,
+  getPSTPDTDateTime,
+} from "@sims/utilities";
+import {
+  createTestingAppModule,
+  describeProcessorRootTest,
+  mockBullJob,
+} from "../../../../../../test/helpers";
+import { ApplicationChangesReportIntegrationScheduler } from "../application-changes-report-integration.scheduler";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeEducationProgramOffering,
+  createFakeStudent,
+  createFakeStudentAssessment,
+  createFakeUser,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import { getUploadedFile } from "@sims/test-utils/mocks";
+import * as Client from "ssh2-sftp-client";
+
+import {
+  Application,
+  ApplicationStatus,
+  AssessmentTriggerType,
+  COEStatus,
+  DisbursementScheduleStatus,
+  EducationProgramOffering,
+  OfferingIntensity,
+  Student,
+  User,
+} from "@sims/sims-db";
+import { createFakeSINValidation } from "@sims/test-utils/factories/sin-validation";
+import { In, IsNull, Not } from "typeorm";
+import {
+  ApplicationChangesReportHeaders,
+  APPLICATION_CHANGES_DATE_TIME_FORMAT,
+} from "@sims/integrations/esdc-integration";
+
+describe(
+  describeProcessorRootTest(QueueNames.ApplicationChangesReportIntegration),
+  () => {
+    let app: INestApplication;
+    let processor: ApplicationChangesReportIntegrationScheduler;
+    let db: E2EDataSources;
+    let sftpClientMock: DeepMocked<Client>;
+    let auditUser: User;
+    let sharedStudent: Student;
+
+    beforeAll(async () => {
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      auditUser = await db.user.save(createFakeUser());
+      const fakeStudent = createFakeStudent();
+      sharedStudent = await saveFakeStudent(db.dataSource, {
+        student: fakeStudent,
+        sinValidation: createFakeSINValidation({ student: fakeStudent }),
+      });
+      // Processor under test.
+      processor = app.get(ApplicationChangesReportIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      db.studentAssessment.update(
+        {
+          previousDateChangedReportedAssessment: { id: Not(IsNull()) },
+        },
+        { previousDateChangedReportedAssessment: { id: null } },
+      );
+    });
+
+    it("Should generate application changes report and update the reported date when there is one or more application changes which are not reported.", async () => {
+      // Arrange
+      // Preparing 2 applications which has application changes to be reported.
+      const firstApplication = await createApplicationChangesData(
+        sharedStudent,
+        1,
+        -1,
+        AssessmentTriggerType.OfferingChange,
+      );
```

Comment:
As suggestion, for better readability I would suggest converting some parameters to options, as below.
```ts
const application = await createApplicationChangesData(
        sharedStudent,
        AssessmentTriggerType.OfferingChange,
        {
          offeringStartDateDifference: 1,
          offeringEndDateDifference: -1,
          isReported: true
        }
      );
```